api-authorization/server/lib/authorization-middleware.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import jwt from 'jsonwebtoken';
+import { ClientError } from './client-error.js';
+
+export function authMiddleware(req, res, next) {
+  /* your code here */
+  try {
+    const auth = req.get('Authorization');
+    const token = auth.split('Bearer ')[1];
```

Comment:
This has a bug. What if there is no Authorization header?

Reply:
I created this "if (!auth || !token) throw new ClientError(401, 'authentication required');" to check if auth and token are defined prior to checking the payload

Reply:
Yes, but you are dereferencing `auth` before you check it. (TypeScript would have caught this for you. Sorry I couldn't figure out a good way to do the auth middleware in TS.)